app/models/article.rb

```diff
@@ -105,4 +108,10 @@ def generate_published_dates
   def downcase_content_format
     self.content_format = self.content_format.downcase
   end
+
+  def create_redirect
+    unless Redirect.where(source_path: short_path, target_path: path).present?
```

Comment:
Here you can do `unless Redirect.exists?(source_path: short_path, target_path, path)`